### DISCUSSION TITLE
error while tryint to upload a google takeout to imich

### DISCUSSION THREAD

**Discussion by nicosalto** [simulot/immich-go#98] [General]
Hi,
i extracted my google takeout into a folder and i am trying to send it to immich, but i get this error and i don't really understand why ?
stat takeout/: no such file or directory

here is the command line
./immich-go -server http://192.168.1.158:30041 -key eRvdYAH88JyxsUkbPlN68  upload -create-album-folder true -google-photos -dry-run smb://192.168.1.158/datas/documents/Pictures/2022 google photo takeout/
immich-go  0.9.0, commit fab5422c386c54fd75ca041dad2324586b587d46, built at 2023-12-09T13:21:10Z

Server status: OK
stat takeout/: no such file or directory

it cant find the takeout folder but i do have it there in that diretory

**Comment by simulot** [comment]
Hi @nicosalto, remove the `/` at the end... I need to improve that!
Did you know, you don't need to unzip takeout files prior importing them using `immiich-go` ?

**Comment by Plamze** [comment]
Hi, 
As I'm having similar problem so suppose that's its ok to use this thread to ask for your guidance.
 After starting the command 
./immich-go -server=http://192.168.50.99:8080 -key <key edited> upload
-create-albums -google-photos //HOMECLOUD/photos/takeout-001.zip

the response that I receive is :

```
Server status: OK
Ask for server's assets...
1628 asset(s) received
Browsing folder(s)...
Done.
Managing albums
Scan of the sources:
     0 files in the input
--------------------------------------------------------
     0 photos
     0 videos
     0 metadata files
     0 files with metadata
     0 discarded files
     0 files having a type not supported
     0 discarded files because in folder failed videos
     0 input total (difference 0)
--------------------------------------------------------
     0 uploaded files on the server
     0 upgraded files on the server
     0 files already on the server
     0 discarded files because of options
     0 discarded files because duplicated in the input
     0 discarded files because server has a better image
     0 errors when uploading
     0 handled total (difference 0)
Done.
bash: -create-albums: command not found
```

The .zip file the I'm trying to upload to immich is 2gb if that matters. trying folder "true" the same result

`bash: -create-albums-folder: command not found`

Trying without this command

`bash: -google-photos: command not found`

**Reply by simulot** [reply]
Try to put everything on a single line
```
./immich-go -server=http://192.168.50.99:8080/ -key upload -create-albums -google-photos //HOMECLOUD/photos/takeout-001.zip
```

And if you have more than one zip, 

```
./immich-go -server=http://192.168.50.99:8080/ -key upload -create-albums -google-photos //HOMECLOUD/photos/takeout-*.zip
```

**Reply by Plamze** [reply]
Unfortunately this returns

```
Server status: OK
stat //HOMECLOUD/photos/takeout-001.zip: no such file or directory
```

**Reply by simulot** [reply]
on windows access to server uses \\\\HOMECLOUD\\photos.... 
I haven't personally tested it.

**Reply by simulot** [reply]
> bash: -create-albums-folder: command not found

Sorry, i have paid attention to this. So you are running immich-go on a linux / mac. The \\ syntaxe is for windows client accessing to SMB servers. 


Try to list the directory where your photos are, and use that path.

**Comment by kricker** [comment]
I am having a similar issue running while trying to import takeout files from Google. I am using the Linux version running on Unraid using a tmux session. I have 45 zip files named "takeout-001.zip" up to "takeout-045.zip". I used the -dry-run feature to see what it would do before I committed to the import. I ran the command:
`./immich-go -server http://192.168.0.100:8888 -key=IMMICHKEY upload -dry-run  -google-photos takeout-*.zip`

But it acts as if there are no files. The files are in the same folder as immich-go without anysubfolders.

![immich-go](https://github.com/simulot/immich-go/assets/114586/32707352-2709-470c-9f24-9a104caa7b97)

The log has:
```
 INFO | Browsing google take out archive... | time="2024-05-25T10:49.37 EDT"
 INFO | 
Input analysis:
---------------------- | time="2024-05-25T10:49.37 EDT"
 INFO | scanned image file                      :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | scanned video file                      :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | scanned sidecar file                    :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | discarded file                          :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | unsupported file                        :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | file duplicated in the input            :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | associated metadata file                :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | missing associated metadata file        :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | 
Uploading:
---------- | time="2024-05-25T10:49.37 EDT"
 INFO | uploaded                                :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | server error                            :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | file not selected                       :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | server's asset upgraded with the input  :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | server has same photo                   :       0 | time="2024-05-25T10:49.37 EDT"
 INFO | server has a better asset               :       0 | time="2024-05-25T10:49.37 EDT"
```

**Reply by simulot** [reply]
It looks like the  takeout-*.zip pattern returns nothing in your context

**Reply by kricker** [reply]
What does that mean? My files are named takeout-001.zip through takeout-045.zip. 

I also tried to do just one file by using the command with takeout-001.zip. That  also had the same result with nothing scanned. It is as if it can't access or find the files. Unfortunately the log doesn't tell me which it is. I've set all the files to full access using chmod 777 thinking it was possibly a read access problem. Yet that did nothing to resolve the issue.

**Reply by simulot** [reply]
I have checked with a bogus name like 
```
./immich-go -server http://192.168.0.100:8888 -key=IMMICHKEY upload -dry-run  -google-photos yolo-*.zip
```
This gives similar result

**Reply by kricker** [reply]
When I try a bogus name like this:

`./immich-go -server http://192.168.0.100:8888 -key=IMMICHKEY upload-dry-run -google-photos test.zip`

I get back:
```
Server status: OK
Connected, user: xxx
open test.zip: no such file or directory:

```
Looking at the folder structure inside of takeout-001.zip I see it like this:

```
takeout-001.zip
  Takeout
    Google Photos
      various folders and .json files
```

**Reply by simulot** [reply]
I meant bogus with a wildcard like: 
```
./immich-go -server http://192.168.0.100:8888 -key=IMMICHKEY upload-dry-run -google-photos test*.zip
```

**Reply by kricker** [reply]
That gives the same result of not scanning any files, but not an error of the files not existing.
<img width="537" alt="Screenshot 2024-05-26 120850" src="https://github.com/simulot/immich-go/assets/114586/ca38e2de-57de-410b-9529-613144874ff2">

And I get the same result if I even try to upload one zip file like this:

`./immich-go -server http://server:port](http://192.168.0.100:8888 -key=IMMICHKEY  upload -dry-run -google-photos takeout-001.zip`

Sorry to be such a pain.

**Reply by kricker** [reply]
So I just tested in a normal terminal session with my unraid server and it appears to be working.
<img width="491" alt="Screenshot 2024-05-26 121607" src="https://github.com/simulot/immich-go/assets/114586/45034e52-1939-42e4-9f37-81c55def2706">


So it has something to do with running immich-go in the tmux session.

**Reply by simulot** [reply]
Good to know. Maybe the fancy mode interferes with  tmux

**Reply by simulot** [reply]
Issue #261

**Reply by kricker** [reply]
I'd be happy to test it for you when you have a new build.

**Reply by simulot** [reply]
It runs using tmux...
![image](https://github.com/simulot/immich-go/assets/5980377/c2f310ad-224a-45b3-a9e0-e118dda91d3d)

**Reply by kricker** [reply]
So it has something to do with my tmux session?

**Reply by simulot** [reply]
the `exit status 1` is returned by the library fails to take the control of the screen. I don't why it doesn't work in your case.
The next version will handle this situation. In the meantime, uses the `-no-ui` option

**Comment by lavcanskiz** [comment]
Hi There,

Sorry I am new to this. I have each of my family members google photos downloaded from google takeout. I ran the string and it returned the numbers of photos and videos, but how can I run the string so it moves each family members google photos to their specific folders in Immich?

This is currently what it looks like in Unraid:
![Screenshot 2024-07-08 204230](https://github.com/simulot/immich-go/assets/87933197/f2105a67-a84b-4fb5-9b4d-f0ceadedfe5d)

Where Path:/import where is mounts to /mnt/user/Media/GooglePhotos, I have 4 other subdirectories for each family member. Do I need to add a custom variable for each one and have it setup like:

/import/familymember1 --> /mnt/user/Media/GooglePhotos/Familymember1
/import/familymember2 --> /mnt/user/Media/GooglePhotos/Familymember2
/import/familymember3 --> /mnt/user/Media/GooglePhotos/Familymember3
/import/familymember4 --> /mnt/user/Media/GooglePhotos/Familymember4

Is there a way when I run the string to import GooglePhotos from their zip files to move them into each family members folder rather than having them in one mixed up folder which currently will go to /import --> /mnt/user/Media/GooglePhotos/?

Here is the string - immich-go -server=http://10.215.5.215:8111 -key=MyAPIKEYGOESHERE upload -dry-run -create-albums -google-photos takeout-*.zip

Is it possible?

Thanks.

**Reply by simulot** [reply]
Hi @lavcanskiz, I don't master immich itself, hence I don't know how to use the import path and the [library feature](https://immich.app/docs/features/libraries). 

Anyway, immich will aggregate all photos  sorted in various paths, into one single collection.
You can liaise with the immich group on (discord)[https://discord.gg/D2MYEqZ6] for help and advices.

That said, Immich can manage several users and keep photos separate. Each user can share albums with others.
If this suits you needs, you have to create 
- on Immich account for each of your users. 
- Get a API key for each of them.
- Use immich-go to import each takeout archive into their respective user's collection, using the user's API key.

**Immich-go can run on your personal computer.** 
Immich-go will browse the takeout files to make remove duplicates, replicate albums, get GPS and dates and upload them into you server using the network. This is simpler than copying files to the server and run immich-go within a container.

For each of your family members, the command would be:
```sh
immich-go -server=http://10.215.5.215:8111/ -key=FAMILYMEMEBERKEY upload -dry-run -create-albums -google-photos /path/to/member/takeout-*.zip
```
Remove the `-dry-run` to load the photos

**Comment by lavcanskiz** [comment]
Thank you for taking the time to respond.

I will reach out to the Discord group and post it there also.

Question, how can I link the API keys to each individual account. All the
users have been created. The API key will work for the upload but then how
do we link the users to that external profile so they can see their
photos and videos?

How can I import the files into the exact folder structure I have setup in
my Unraid Server?

I will run the import with my photos and videos only and see how the path
is created and see if I can then link the external libraries to each
profile I guess.

Thanks.


Kind Regards,


*Zlatko Lavcanski*

E: ***@***.***

M: +61 4 31 751 323

**Reply by simulot** [reply]
It's possible to just import the takeout archives as an external library. But you'll end up with lot of duplicates and will not take advantage of JSON metadata files.

Immich-go does the job in a different way. It browses the takeout files to build a consistent set of photos with metadata taken from JSONs and it upload them into the user's collection, user determined by the API key given on the command line. The Immich server will stores the photos in its own way, optimized for its purpose.
